src/preview/index.js

```diff
@@ -56,10 +56,10 @@ export default class Preview {
     return () => {
       let webUrl = null;
       let channel = addons.getChannel();
-      if (!channel) {
+      if (params.resetStorybook || !channel) {
         const host = params.host || 'localhost';
         const port = params.port || 7007;
-        const url = `ws://${host}:${port}`;
+        const url = `ws://${host}:${port}/${params.query}`;
```

Comment:
If `params.query` is not set, the url will have `/undefined` in the end. Better set it to an empty string if it's not set.

```
const query = params.query || '';
const url = `ws://${host}:${port}/${query}`;
```